deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/TokenCredentialType.java

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.sdk.iot.deps.auth;
+
+/**
+ * An enumeration of supported authorization methods for service client connections.
+ */
+public enum TokenCredentialType
+{
+    /**
+     * Authorize with a shared access signature.
+     */
```

Comment:
Is this enum for just the sastoken type? Or did I miss something.

Reply:
Right now, yes. The idea is to add another enum value here for RBAC once that is ready from the service side

Reply:
Edit on this comment: I'm opting to use an azure-core-amqp enum instead of creating this enum now. In azure-core-amqp, they have a class just like this, but with RBAC as the second enum value. I've deleted this local enum now